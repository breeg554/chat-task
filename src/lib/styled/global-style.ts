import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  html,input{
    font-family: 'Roboto', sans-serif;
    font-size: 10px;   //1rem = 10px
  }
  body,button,input{
    background-color: #fff;
    color: ${({ theme }) => theme.colors.core.primary}
  }

`;
