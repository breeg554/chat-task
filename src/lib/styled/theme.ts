export type ThemeType = typeof theme;

export const theme = {
  colors: {
    core: {
      primary: "#2C3E50",
      secondary: "#555555",
      tertiary: "rgba(0,0,0,0.25)",
      success: "#74DD88",
    },
    hover: {
      primary: "#222",
      secondary: "#444",
      tertiary: "rgba(0,0,0,0.4)",
      success: "#95ee99",
    },
  },
  gradients: {
    core: {
      primary: "linear-gradient(to right, #00c6ff, #0072ff)",
    },
    hover: {
      primary: "linear-gradient(to right, #00a5dd, #0061dd)",
    },
  },
};
