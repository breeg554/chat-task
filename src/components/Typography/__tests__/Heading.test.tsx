import { render } from "@testing-library/react";
import { Heading } from "..";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../lib/styled";

describe("Heading component", () => {
  it("renders out and match the snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Heading>Heading</Heading>
      </ThemeProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
