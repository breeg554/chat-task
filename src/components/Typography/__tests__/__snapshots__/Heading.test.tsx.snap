// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component renders out and match the snapshot 1`] = `
<div>
  <h1
    class="sc-bdvvtL lkBEZa"
  >
    Heading
  </h1>
</div>
`;
